src/strands/session/agent_session_manager.py

```diff
@@ -0,0 +1,102 @@
+"""Agent session manager implementation."""
+
+import logging
+
+from ..agent.agent import _DEFAULT_AGENT_ID, Agent
+from ..agent.state import AgentState
+from ..types.content import Message
+from ..types.exceptions import SessionException
+from ..types.session import (
+    Session,
+    SessionAgent,
+    SessionMessage,
+    SessionType,
+)
+from .session_manager import SessionManager
+from .session_repository import SessionRepository
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SESSION_AGENT_ID = "default"
+
+
+class AgentSessionManager(SessionManager):
```

Comment:
In one of our discussions we talked about wanting updates to be persisted immediately after they occurred.  It might be good to document/detail that somewhere just to keep them in mind going forward.  

(I did something similar with hooks @ https://github.com/strands-agents/sdk-python/blob/5dc3f594bf9da5a72a2f15b15fe49508efb2e513/src/strands/hooks/rules.md#L4)